packages/template-retail-react-app/app/components/display-price/index.jsx

```diff
@@ -7,61 +7,160 @@
 
 import React from 'react'
 import PropTypes from 'prop-types'
-import {Skeleton, Text} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {Box, Text} from '@salesforce/retail-react-app/app/components/shared/ui'
 import {useIntl} from 'react-intl'
-import {useCurrency} from '@salesforce/retail-react-app/app/hooks'
 
-const DisplayPrice = ({
-    basePrice,
-    discountPrice,
-    isProductASet = false,
-    currency,
-    discountPriceProps,
-    basePriceProps,
-    skeletonProps
-}) => {
+/**
+ * @param priceData - price info extracted from a product
+ * // If a product is a set,
+ *      on PLP, Show From X where X is the lowest of its children
+ *      on PDP, Show From X where X is the lowest of its children and
+ *          the set children will have it own price as From X (cross) Y
+ * // if a product is a master
+ *      on PLP and PDP, show From X (cross) Y , the X and Y are
+ *          sale and list price of variant that has the lowest price (including promotionalPrice)
+ * // if a standard/bundle
+ *      show exact price on PLP and PDP as X (cross) Y
+ * @param currency - currency
+ */
+const DisplayPrice = ({priceData, currency}) => {
     const intl = useIntl()
-    const {currency: activeCurrency} = useCurrency()
-    return (
-        <Skeleton isLoaded={basePrice} display={'flex'} {...skeletonProps}>
-            <Text fontWeight="bold" fontSize="md" mr={1}>
-                {isProductASet &&
-                    `${intl.formatMessage({
-                        id: 'product_view.label.starting_at_price',
-                        defaultMessage: 'Starting at'
-                    })} `}
+    const {listPrice, salePrice, isASet, isMaster, isOnSale, isRange, hasRepresentedProduct} =
+        priceData
+    const salePriceText = intl.formatNumber(salePrice, {
+        style: 'currency',
+        currency: currency
+    })
+    const listPriceText =
+        listPrice &&
+        intl.formatNumber(listPrice, {
+            style: 'currency',
+            currency: currency
+        })
+
+    const prefixText = intl.formatMessage({
+        id: 'price_display.label.from',
+        defaultMessage: 'From '
+    })
+
+    const ariaLabelSalePrice = intl.formatMessage(
+        {
+            id: 'display_price.assistive_msg.current_price',
+            defaultMessage: `current price {price}`
+        },
+        {
+            price: intl.formatNumber(salePrice || 0, {style: 'currency', currency})
+        }
+    )
+    const ariaLabelListPrice = intl.formatMessage(
+        {
+            id: 'display_price.assistive_msg.strikethrough_price',
+            defaultMessage: `old price {price}`
+        },
```

Comment:
Minor: how about calling it _original_ price?
```suggestion
    const ariaLabelListPrice = intl.formatMessage(
        {
            id: 'display_price.assistive_msg.strikethrough_price',
            defaultMessage: `original price {price}`
        },
```